core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1391,10 +1391,11 @@ package object config {
 
   private[spark] val SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED =
     ConfigBuilder("spark.shuffle.readHostLocalDisk")
-      .doc(s"If enabled (and `${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key}` is disabled and external " +
-        s"shuffle `${SHUFFLE_SERVICE_ENABLED.key}` is enabled), shuffle " +
-        "blocks requested from those block managers which are running on the same host are read " +
-        "from the disk directly instead of being fetched as remote blocks over the network.")
+      .doc(s"If enabled (and `${SHUFFLE_USE_OLD_FETCH_PROTOCOL.key}` is disabled and 1) external " +
+        s"shuffle `${SHUFFLE_SERVICE_ENABLED.key}` is enabled or 2) ${DYN_ALLOCATION_ENABLED.key}" +
+        s" is disabled), shuffle blocks requested from those block managers which are running on " +
+        s"the same host are read from the disk directly instead of being fetched as remote blocks" +
+        s" over the network.")
```

Comment:
Maybe add some extra words around this depends on having non-isolated container storage (e.g. a shared hostPath in the k8s world)

Reply:
Do you mean, in k8s, this feature should only work when executors are using non-isolated container storage?

Reply:
How would it work with isolated storage?

Reply:
I'm not familiar with cloud env. So I just ask for your confirmation.